src/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -6,28 +6,57 @@
 #include "software/ai/hl/stp/tactic/primitive.h"
 #include "software/logger/logger.h"
 #include "software/util/typename/typename.h"
+template<class Tactic_FSM>
```

Comment:
usually if you have to define functions with templates in the header file: https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl